Scala 2.13 does not compile with Gradle on Windows

The [original bug](https://github.com/gradle/gradle/issues/9857) is reported in the Gradle repository, but I believe it should be brought into attention in this repository as well.

## steps

Use Gradle to compile Scala 2.13.0 project.

## problem

```
Compiling with Zinc Scala compiler.
'compiler-interface-2.13.0-55.0.jar' not yet compiled for Scala 2.13.0. Compiling...
C:\Users\petzold\AppData\Local\Temp\sbt_76a43d36\xsbt\Compat.scala:152: error: postfix operator headOption needs to be enabled
by making the implicit value scala.language.postfixOps visible.
This can be achieved by adding the import clause 'import scala.language.postfixOps'
or by setting the compiler option -language:postfixOps.
See the Scaladoc for value scala.language.postfixOps for a discussion
why the feature needs to be explicitly enabled.
} headOption
^
one error found
```

## expectation

compiles.

## notes

As I described in one of the comments on the original issue, I suspect that [this Scala change](https://github.com/scala/scala/pull/6559) is responsible for Gradle's inability to compile Scala 2.13, as this change makes Scala 2.13 incompatible with SBT 0.13.15, which is what Gradle uses for compilation.

Perhaps all it takes is to build and publish the [old Zinc project](https://github.com/typesafehub/zinc) with SBT 0.13.18. I was able to do it locally, but I can't really be sure I've done it properly as I'm quite unfamiliar with this project.

Furthermore, there might be other Scala 2.13 changes which are incompatible with SBT 0.13.18.

--- a/src/compiler/scala/tools/nsc/Reporting.scala
+++ b/src/compiler/scala/tools/nsc/Reporting.scala
@@ -115,7 +115,9 @@ trait Reporting extends scala.reflect.internal.Reporting { self: ast.Positions w
       val msg = s"$featureDesc $req be enabled\nby making the implicit value $fqname visible.$explain" replace ("#", construct)
       // don't error on postfix in pre-0.13.18 xsbt/Compat.scala
       def isSbtCompat =
-        (featureName == "postfixOps" && pos.source.path.endsWith("/xsbt/Compat.scala") && Thread.currentThread.getStackTrace.exists(_.getClassName.startsWith("sbt.")))
+        (featureName == "postfixOps" &&
+        (pos.source.path.endsWith("/xsbt/Compat.scala") || pos.source.path.endsWith("""\xsbt\Compat.scala""")) &&
+        Thread.currentThread.getStackTrace.exists(_.getClassName.startsWith("sbt.")))
       if (required && !isSbtCompat) {
         reporter.error(pos, msg)
       } else featureWarning(pos, msg)

